### PR TITLE
hal_silabs: Add support for librail for 1x devices

### PR DESCRIPTION
**Pull request by yonsch**
Update hal_silabs to support the use of librail for efr32xg1x devices.

### CODE REVIEW

**Comment by yonsch** [comment]
ref: https://github.com/zephyrproject-rtos/hal_silabs/pull/35

**Comment by fkokosinski** [comment]
Hi, thanks for this PR. It looks like:
1. you need to create "binary blobs integration" issue using the template from here: https://github.com/zephyrproject-rtos/zephyr/issues/new/choose
2. no zephyr platform uses these blobs right now (no bluetooth changes etc.)?

**Comment by yonsch** [comment]
@fkokosinski Thanks, the issue is here: #61109 
I'm using RAIL without BT, so I haven't added it (I don't have a way to test it). I can add it if someone is willing to test it, or it can be added later.

**Comment by fkokosinski** [comment]
> @fkokosinski Thanks, the issue is here: #61109

Thanks!

> I'm using RAIL without BT, so I haven't added it (I don't have a way to test it). I can add it if someone is willing to test it, or it can be added later.

What about `CONFIG_SOC_GECKO_USE_RAIL`? Do you not need to set it?

**Comment by yonsch** [comment]
> What about CONFIG_SOC_GECKO_USE_RAIL? Do you not need to set it?

Yes, I do it in my `prj.conf`.
Also, two questions:
1. I see that the SDK already has a 4.3 version, is there any reason not to use it?
2. Currently the URLs in `module.yml` reference a branch and not a commit/tag. Wouldn't that be a problem if a newer version is pushed to that branch by Silabs?

**Comment by fkokosinski** [comment]
> > What about CONFIG_SOC_GECKO_USE_RAIL? Do you not need to set it?
> 
> Yes, I do it in my `prj.conf`. Also, two questions:
> 
>     1. I see that the SDK already has a 4.3 version, is there any reason not to use it?

Not really. It just that newer blobs weren't tested with the BT driver.

>     2. Currently the URLs in `module.yml` reference a branch and not a commit/tag. Wouldn't that be a problem if a newer version is pushed to that branch by Silabs?

Good catch. We have SHAs of those blobs, so we can identify them, but indeed - it would be a if SiLabs were to push a newer version to that branch. Let's change URLs to point to tags instead.

**Comment by yonsch** [comment]
Update: fixed FPU issue for series 1 devices and found out that pinctrl was not implemented for most of them. I've added an implementation for EFR32FG13P (BRD4255A which is what I have).

File: boards/arm/efr32_radio/Kconfig.defconfig [hunk 1]
```diff
@@ -32,6 +32,7 @@ config LOG_BACKEND_SWO_FREQ_HZ
 if SOC_GECKO_USE_RAIL
 
 config FPU
+	default n if SOC_GECKO_SERIES1
```
Comment (fkokosinski) [inline boards/arm/efr32_radio/Kconfig.defconfig:35]: Hmmm, we don't need to disable FPU if we're not using RAIL. So we should check if we're `SOC_GECKO_SERIES1` and if we're using RAIL here IMO.

Comment (yonsch) [reply boards/arm/efr32_radio/Kconfig.defconfig:35]: This is already checked on line 32

Comment (fkokosinski) [reply boards/arm/efr32_radio/Kconfig.defconfig:35]: Monday mornings ;) You're correct, thanks

**Comment by fkokosinski** [comment]
Hi @yonsch! I've merged https://github.com/zephyrproject-rtos/hal_silabs/pull/35. Please update the HAL revision that this PR points to to the newest one. Thanks!

**Review by fkokosinski — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
hey @yonsch looks like this broke some boards:

```
$ west build -p -b efm32wg_stk3800 samples/basic/blinky
...
CMake Error at /usr/local/google/home/fabiobaltieri/zephyrproject/modules/hal/silabs/gecko/CMakeLists.txt:54 (message):
  Unknown gecko series type.  Is CONFIG_SOC_GECKO_SERIES1/2 set?
```

could you take a look?

**Pull request merged**